service/raft/group0_voter_handler.cc

```diff
@@ -496,6 +496,23 @@ future<> group0_voter_handler::update_nodes(
 
```

Comment:
The new version of this commit fixes another bug: `update_nodes` ignored leaving voters, which could result in having 6 voters for example. You should mention it in the commit message.

I would also consider changing the commit message and the PR title to something more general like "raft: fix voter assignment of transitioning nodes".

We could add a regression test for this bug, but I think it's an overkill.